### ISSUE TITLE
Musescore patched this I think

### ISSUE THREAD

**Issue by dayoshiguy** [Xmader/musescore-downloader#48]
It says I have a banned plugin and that they very angery lol

**Comment by dayoshiguy** [comment]
![Screenshot 2020-11-05 at 3 45 49 PM](https://user-images.githubusercontent.com/11911537/98294478-04b14980-1f7e-11eb-863f-e552f0d4aa71.png)

**Comment by dayoshiguy** [comment]
what tf is the point of musescore if they pull shit like this... They want us to just stare at the sheets and not use them? wtf?

**Comment by Apeii050** [comment]
I got this today too. Should I be concerned about the whole reported to law enforcement thing?

**Comment by Xmader** [comment]
> I got this today too. Should I be concerned about the whole reported to law enforcement thing? I uninstalled just in case but is there any confirmation that its an actual threat?
> 
> Musescore sucks now tbh

They are just scaring you to make you stop using this script.

**Comment by Apeii050** [comment]
Well, I dunno. Considering they are partnered up with some serious companies and google knows everything, so I don't reckon I want to risk it, mainly because getting involved in legal issues would be very bad for me atm. I think your script is awesome, so keep at it, but I don't reckon I'll be using it until I get a little more clarification on the issue. 
I suppose I can only hope for the best at this point.

**Comment by Xmader** [comment]
fixed in `v0.11.2`

**Comment by popyd10** [comment]
The "Unauthorized use of copyright content" message appears a few seconds after page loading even after new update.

**Comment by Apeii050** [comment]
Just out of curiosity, can they actually send police? I having anxiety about the whole things, silly as that may sound. This issue sounds pretty legit. Whether it's okay or not, the law is the law and this plugin does seem to breech it.

Also, Xmader, I noticed you had a Musescore alternative website, wont that eventually suffer the same legal issues Musescore had to face?

**Comment by SealsRock12** [comment]
> Just out of curiosity, can they actually send police? I having anxiety about the whole things, silly as that may sound. This issue sounds pretty legit. Whether it's okay or not, the law is the law and this plugin does seem to breech it.
> 
> Also, Xmader, I noticed you had a Musescore alternative website, wont that eventually suffer the same legal issues Musescore had to face?

I don't think so. This is because the musescore software itself is under the GPL license, which allows others to do almost anything with it. Even though the musescore dot com website is using proprietary code, Xmander is completely allowed to make their alternative as long as they aren't using any of musescore dot com's code.

**Comment by Apeii050** [comment]
> I don't think so. This is because the musescore software itself is under the GPL license, which allows others to do almost anything with it. Even though the musescore dot com website is using proprietary code, Xmander is completely allowed to make their alternative as long as they aren't using any of musescore dot com's code.

Code isn't the issue here, it's the songs themselves. This script gives access to downloading the files, which according to Musescore is breaking the law. And when I am talking about Xmaders alternative, I mean legal issues such as big music companies demanding compensation for infringement.

**Comment by Xmader** [comment]
There are no such "copyrighted contents" at all.

As you know, scores on musescore.com are all user uploaded. 
(Except for the newly added "official scores", which this script cannot cover #40. They are actually from UltimateGuitar's site)

According to Musescore's term of use (<https://web.archive.org/web/20200707121931/https://musescore.com/legal/terms>), "Musescore does not claim any ownership rights in User Generated Content…", 
Musescore does not hold any copyrights.

**Comment by Apeii050** [comment]
MuseScore doesn't hold the copyrights, however, they are responsible for maintaining the legal stuff concerning songs on their website with copyrighted content. It is for this reason that I believe their threat is serious.

**Comment by SealsRock12** [comment]
> > I don't think so. This is because the musescore software itself is under the GPL license, which allows others to do almost anything with it. Even though the musescore dot com website is using proprietary code, Xmander is completely allowed to make their alternative as long as they aren't using any of musescore dot com's code.
> 
> Code isn't the issue here, it's the songs themselves. This script gives access to downloading the files, which according to Musescore is breaking the law. And when I am talking about Xmaders alternative, I mean legal issues such as big music companies demanding compensation for infringement.

Sorry, I was talking about LibreScore, the musescore alternative that @Xmander is creating. To my knowledge, LibreScore isn't hosting any copyrighted songs nor taking anything from musescore dot com.

> There are no such "copyrighted contents" at all.
> 
> As you know, scores on musescore.com are all user uploaded.
> (Except for the newly added "official scores", which this script cannot cover #40. They are actually from UltimateGuitar's site)
> 
> According to Musescore's term of use (https://web.archive.org/web/20200707121931/https://musescore.com/legal/terms), "Musescore does not claim any ownership rights in User Generated Content…",
> Musescore does not hold any copyrights.

They can take down the script by a cease and desist under the anti-circumvention law.

**Comment by Brokop** [comment]
I am personally not using this script, I am just following the development and situation here. Based on my knowledge, most of MuseScore website threats are just threats which are very poorly put together to scare people from using any extensions. Based on my limited knowledge, saying the content is belonging to big companies and that they pay royalties to all the original scores uploaded by small individuals is pretty much false. Also many songs made by people are actually under licenses which should not allow MuseScore to monetize them at all. However MuseScore still does so by forcing users to pay for downloading it and by advertising their Pro subscription alongside it. Saying you will get reported to local police cant do them any good and in my opinion should backfire sooner or later. Also DCMA shouldnt apply to this extension code. If there will be a discord or some group for the new LibreScore website to rival MuseScore I would gladly pop right in. I hope this threatening is just a doing of few unsupervised people at MuseScore rather than entire company decision.

**Comment by org250** [comment]
I have a law degree, so I can enlighten you a little bit on this issue. The rights to the works of authors belong to the author of the work. This applies to both the music record and sheet music. Therefore it doesn't matter if the content is user-generated or not, the rights to a piece of music still belong to the author of the work, not to the transcribers of sheet music. If MuseScore does have agreements with copyright holders, then using this script may raise legal issues for users.

**Comment by Xmader** [comment]
Musescore cannot send a DMCA notice isn't it? because Musescore is not the copyright holder.
Musescore would only be licensed to use them on the site (just like anyone buys a licence), not permitted to deal with the copyright things on behalf of the copyright holders.

**Comment by Neiko-Real** [comment]
still not fixed.
still same thing

**Comment by Neiko-Real** [comment]
@Xmader Unfortunately under the DMCA act, if only IF they have signed agreements WITH SPECIFIC AUTHORS of said musics they can invoke this DMCA rule. 
They do not have the agreements of people like CG5 and the living tombstone (FNAF) music. 
They cannot invoke the DMCA For those. But for famous people like Ed Sheeran and Shawn Mendes etc this copyright rule CAN be invoked

**Comment by Neiko-Real** [comment]
@Xmader They should only block authors with agreements to them.
If they sell material without the permission of ALL and I SAID ALL authors of the sheet music in musescore, they can be put under legal battle for that.
And Finally. if this "public domain scores and original works may still be downloaded for free"
These may be free BUT, could they assure that they got the permission from all the authors that the said users transcribed the sheet music onto.
AND even with the said PRO subscription, They allow DOWNLOADING OF MIDI FILES which means people COULD COPY them.
Therefore making them useless for the transcribers, and the SAID "copyright holders"

**Comment by Neiko-Real** [comment]
keybase.io group ready. @Xmader sign up with your email i@xmader.com
you will be granted owner on joining

**Comment by Neiko-Real** [comment]
everyone else send their email to dilanuthan66@gmail.com

**Comment by Neiko-Real** [comment]
i will admit people

**Comment by SealsRock12** [comment]
> I have a law degree, so I can enlighten you a little bit on this issue. The rights to the works of authors belong to the author of the work. This applies to both the music record and sheet music. Therefore it doesn't matter if the content is user-generated or not, the rights to a piece of music still belong to the author of the work, not to the transcribers of sheet music. If musescore does have agreements with copyright holders, then using this script may raise legal issues for users.

This is why I am not using the script either, just following the development of it. Although if Musescore does have agreements with copyright holders, it is the respective copyright holder's jobs to file a lawsuit or a DMCA notice against this script.

**Comment by Neiko-Real** [comment]
It is fine, since they cannot physically arrest us

**Comment by Apeii050** [comment]
Whats to stop them from doing so?

**Comment by Neiko-Real** [comment]
hmm
for one the PR failure. HEADLINES: MUSECORE ARRESTING FREE MUSIC USERS

**Comment by Neiko-Real** [comment]
hmmmmm???

**Comment by Neiko-Real** [comment]
your killin me, XD. Arrested for downloading music

**Comment by Neiko-Real** [comment]
imagine a 13 year old, get arrested by police for THIS

**Comment by Neiko-Real** [comment]
theres probably children using this script who knows??

**Comment by Brokop** [comment]
_"12 Year old college student who wanted to practice his favorite Living Tombstone track in Synthesia facing charges by MuseScore for allegedly downloading music."_

**Comment by Brokop** [comment]
Besides, what data really museScore collects? They can get approximate location by IP, but thats about where it ends. Its not like they are putting their resources into an entire tracking investigation. Imagine using Onion Relay so you can access MuseScore

**Comment by Neiko-Real** [comment]
@Brokop exactly MY POINT

**Comment by Neiko-Real** [comment]
you know really its unfair

**Comment by Brokop** [comment]
I wonder what would the MuseScore open-source OGs App PR team say on these threats the web department mods are doing. Still cant understand how a considerably big site like MuseScore has so unprofessional and may I say nearly comedic threats. If you are on youtube and actually use any copyrighted content, you are just either forced to mute it or get one out of three strikes. But MuseScore be like: Hold on kiddo, I see you got this very bad software I dont like. Doesnt matter whether you actually used it illegaly. You gettin reported to police bucko. Yeeah. What do you mean it aint illegal? It illegal cuz I represent big compan.

**Comment by Apeii050** [comment]
I was thinking more along the lines of, “Musescore users utilize script to download and redistribute music”. Whether we actually are redistributing or not, they could twist the story to be whatever they wanted, especially since musescore has big music companies behind it. UG is pretty scummy from what I know and i wouldn't put it past them to pull something like that. Plus, some users made their accounts with google, so if Musescore, or UG, is really bent on screwing people over, google could probably provide them with extensive data.

**Comment by Neiko-Real** [comment]
lol

**Comment by Neiko-Real** [comment]
cant pro users 'redustribute'?

**Comment by Brokop** [comment]
YEA OF COURSE! YOU PAY US SO WE LOOK PAST ALL THE COPYRIGHT.

**Comment by Brokop** [comment]
MuseScore would have to get a warrant first from law enforcement to get any data outta google or ISP.

**Comment by Neiko-Real** [comment]
yep, no

**Comment by Neiko-Real** [comment]
when visiting the website they could get our ips

**Comment by Apeii050** [comment]
Couldn’t they get the warrants if there is copyright infringement going on?

**Comment by Brokop** [comment]
Public IPs will just point to the general city geodata.

**Comment by Brokop** [comment]
And you cant get any dirt on your for just owning this extension at all. Thats a fact as long as this extension doesnt have any copyrightable data included. And as far as I know its based on previously public API which hasnt even been altered in a way to make this illegal.

**Comment by Neiko-Real** [comment]
yep. exactly...

**Comment by Neiko-Real** [comment]
its all buullcrap

**Comment by Brokop** [comment]
They would have to make sure there is an infrindement going on in the first place and very extensively prove it.

**Comment by Apeii050** [comment]
But usage of the extension could be found through files on your pc, unless you do a hard pc wipe

Whether or not they actually can take you to jail, i’d prefer not to be involved in anything legal.

**Comment by Neiko-Real** [comment]
hmm

**Comment by Brokop** [comment]
Hi @Apeii050 
MuseScore as a site has nearly no clue where you are, what PC you are using or who you are. Only law enforcement is authorized to come and check it out only with a valid warrant given by some justice dept. I advise you to not use the extension to download anything illegal but I doubt anything would actually happen. Its likesaying  just owning Torrent client is illegal or accessing TOR is illegal.

**Comment by Brokop** [comment]
Not to mention warrant and stuff would be costly for MuseScore. And It would also drag attention to some shady licensing issues MuseScore has about commercial use of the content users upload under licenses which afaik prohibit that...

**Comment by Brokop** [comment]
MuseScore Website is currently trying to scare everyone and close this extension off. I wouldnt do any fuzz about it if their attitude wouldnt be so unprofessional and wouldnt sound like they have no clue what they are talking about

**Comment by Apeii050** [comment]
I realize I am probably over exaggerating some of this, but my thoughts lead me to conclude that legal issues are possible. Guess I’m being driven by fear, but I also just can’t get into legal issues rn, no money to pay the fines

**Comment by Neiko-Real** [comment]
dude there wont be A FINE

**Comment by Brokop** [comment]
Just dont acknowledge you used it for anything illegal or at all @Apeii050, which is obviously the truth here as you were saying. otherwise they cant do pretty much anything. And I am very exaggerating what they can do.

**Comment by Neiko-Real** [comment]
the courts wont even accept the case

**Comment by Neiko-Real** [comment]
imagine the courts allowed this xD

**Comment by Brokop** [comment]
Yeah. I bet there wont even be any in the first place. This entire threat is ridiculous.

**Comment by Neiko-Real** [comment]
yep

**Comment by Neiko-Real** [comment]
so dont even think about a "possiblity"

**Comment by Brokop** [comment]
Just as I said, as long as you dont march to their HQ, give them your number and start spamming their support about illegaly using the actually licensed content... there is really nothing much happening.

**Comment by Apeii050** [comment]
Why cant they fine you over downloading arrangements and scores from other people? If I downloaded say Mario theme arrangement, couldn’t I be fined since the rights belong to nintendo?

**Comment by Neiko-Real** [comment]
lol

**Comment by Neiko-Real** [comment]
nop, since many people posted them ON YOUTUBE for free....

**Comment by Brokop** [comment]
Couldnt museScore be fined for possesing a transcription of Mario copyrighted music which has not been acknowledged by nintendo for being derivative?

**Comment by Neiko-Real** [comment]
and didnt get copyright??????????

**Comment by Neiko-Real** [comment]
i bet musescore dosent even have the permission of the authors

**Comment by Brokop** [comment]
I think there is some passage in TOS about uploading stuff you dont own but MuseScore doesnt really give a crap about moderating that

**Comment by Neiko-Real** [comment]
yeppppppppppppppppppp

**Comment by Neiko-Real** [comment]
ximich looks ugly

**Comment by Neiko-Real** [comment]
just added that for lolz

**Comment by Brokop** [comment]
So just imagine them actually spending a month of research, hundreds of dollard on a case file report and getting involved with international polices just for this. not to mention they would lose the case.

**Comment by Brokop** [comment]
Well now lets dont get toxic at anyone here. I am trying to be unbiased and decent here.

**Comment by Brokop** [comment]
do we have any actual confirmation @ximich has any affiliation with MuseScore?

**Comment by Brokop** [comment]
Also a fun fact here, many big shops use copyrighted music as background music without any approval from the copyright holder. And these can be big companies.

**Comment by Neiko-Real** [comment]
yep, he is either admin or dev, but i think he needs english lessons

**Comment by Neiko-Real** [comment]
too much spelling mistakes

**Comment by Brokop** [comment]
I think he more necessarily needs better understanding of how law and copyright works. Didnt @Ximich actually say something like we are all getting our data sent to police just for being here in this thread?

**Comment by SealsRock12** [comment]
Yes, people like me who don't even use this can't get into any legal trouble. We aren't helping the development of the script, either, just discussing it.

**Comment by Neiko-Real** [comment]
um no

**Comment by Neiko-Real** [comment]
why are you discussing if you arent suppporting?

**Comment by Brokop** [comment]
And as I said, this script uses previously public API so the script itself isnt defyng any legal issues whatsoever.

**Comment by Neiko-Real** [comment]
nop he is just a idiot

**Comment by Neiko-Real** [comment]
grasping at straws

**Comment by Brokop** [comment]
The API was released and as we all know, once its released you cant magically change its copyright status to anyone already obtaining the copy.

**Comment by SealsRock12** [comment]
I'm not writing any code, so how am I supporting this?

**Comment by Brokop** [comment]
Yeah, @SealsRock12 hasnt pushed or merged any code. We are just joining the discussion here. I find the LibreScore project interesting too.

**Comment by Neiko-Real** [comment]
your support is not in CODE

**Comment by Brokop** [comment]
@Neiko-Real, legally speaking he isnt participating in development. @SealsRock12 is just cautious to not be associated in the case MuseScore MPs SWAT everyone here hehe. Which is understandable.

**Comment by Neiko-Real** [comment]
yea, our world will end unless we praise musescore

**Comment by Brokop** [comment]
The MuseScore Copyright Insurgency will destroy the opposition.

**Comment by Neiko-Real** [comment]
no
WE ARE  REBELS

**Comment by Brokop** [comment]
Speaking of MuseScore, they disabled downloading of MIDIs over a single day for all non public-domain scores. For everyone. Every score. They put em all behind a paywall. This probably could lead to a shady copyright issue much bigger than some guy downloading a midi. MuseScore could face actual charges if someone would pu the effort into it.

**Comment by Brokop** [comment]
Look I think we shouldnt be spamming in a commit thread like this. Is there a discord server planned for this project I can join?

**Comment by Neiko-Real** [comment]
yea ill make one soon

**Comment by Xmader** [comment]
> still not fixed.
> still same thing

fixed again in `v0.11.4`

**Comment by dayoshiguy** [comment]
someone make a youtube video about this musescore needs to be held responsible for this bs

**Comment by Neiko-Real** [comment]
Just comment your keybase username
Ill make a discord

**Comment by Neiko-Real** [comment]
https://discord.gg/MmYkkkHzfE first 10 can join
@Xmader if you can please join
When you have time of course 🙂

**Comment by Neiko-Real** [comment]
https://discord.gg/MmYkkkHzfE
Here you go join

**Comment by North-West-Wind** [comment]
Hey @Xmader I think they patched it again. However, it seems that they only removed the GUI when extension is detected (By removed I mean it turns into the "Unauthorized use of Copyrighted Content" thing).

The detection takes a few seconds and I was quick enough to click the "Download MSCZ" button, and although the preview is gone, the script can still run in the background.

**Comment by TheAlienDrew** [comment]
I found something out, if you just take the code and run it later after the website is finished loading the script will run fine.

I suggest maybe having the script run after a timeout of maybe 5 to 10 seconds... I'll do some more testing and publish a merge pull of my suggested changes.

**Comment by TheAlienDrew** [comment]
See my pull https://github.com/Xmader/musescore-downloader/pull/49... the script works again, but must wait 8 seconds at the least so far for it to work again without tripping Musescore.

**Comment by marcan** [comment]
If you use such a script and it violates the website T&C's, they could hypothetically sue you. They could also hypothetically sue you for copyright infringement if they have agreements with publishers. They could sue you for any made up reason whatsoever, because anyone can sue anyone else (especially in the US).

They cannot, however, send the police to your door. At least not in most jurisdictions.

Practically speaking, it's blatantly obvious that the warning is an amateurish attempt at scaring people, and that no actual competent lawyers are running this operation at MuseScore.

**Comment by TheAlienDrew** [comment]
They especially can't do anything against people that aren't redistributing the stuff they download through this extension.

**Comment by TheAlienDrew** [comment]
In any case though, @Xmader not sure if this will help at all, as I'm not the best at reverse engineering... but I believe I found the part in Musescore's source code that detects when your downloader is loaded... note the zip file only contains the one JS file, but it 'talks' with other files when doing the detection. I did however highlight and include a note of where such assumed detection is taking place.

https://anonfiles.com/v3369fn9p5/Musescore_s_Downloader_Detection_in_JS_source_zip

**Comment by TheAlienDrew** [comment]
As for the other files you might want to look at are: ms~runtime..........js, system...........js, and maybe the vendor........js

(......... imply the strings that always change)

**Comment by TheAlienDrew** [comment]
Wait nevermind, I see you already know that when taking a look at https://github.com/Xmader/musescore-downloader/blob/master/src/btn.ts

**Comment by Xmader** [comment]
fixed again in `v0.11.6`

**Comment by TheAlienDrew** [comment]
Just tried the update, it's not working for me.

**Comment by Xmader** [comment]
> Just tried the update, it's not working for me.

It works on Firefox. Fixing on Chrome

**Comment by Xmader** [comment]
fixed in `v0.12.0`

**Comment by djvplayz** [comment]
and it's back

**Comment by Jabster28** [comment]
@Xmader maybe write a CI function that changes the code every half an hour so musescore's devs would have a hard time? ;)

**Comment by Neiko-Real** [comment]
@Xmader have you joined. I see almeone with your name

**Comment by Neiko-Real** [comment]
New invite link
https://discord.gg/DKu7cUZ4XQ

**Comment by Xmader** [comment]
> @Xmader have you joined. I see almeone with your name

Yes `Xmader#9675`
